test/test_blobstore.go

```diff
@@ -0,0 +1,84 @@
+package main
+
+import (
+	"errors"
+	"net/http"
+	"net/url"
+	"os"
+
+	c "github.com/flynn/flynn/Godeps/_workspace/src/github.com/flynn/go-check"
+)
+
+type BlobstoreSuite struct {
+	Helper
+}
+
+var _ = c.Suite(&BlobstoreSuite{})
+
+func (s *BlobstoreSuite) TestBlobstoreBackendSwitching(t *c.C) {
+	s3Config := os.Getenv("BLOBSTORE_S3_CONFIG")
+	if s3Config == "" {
+		// BLOBSTORE_S3_CONFIG should be set to a valid configuration like:
+		// backend=s3 access_key_id=xxx secret_access_key=xxx bucket=blobstore-ci region=us-east-1
+		t.Skip("missing BLOBSTORE_S3_CONFIG env var")
+	}
+
+	r := s.newGitRepo(t, "http")
+	t.Assert(r.flynn("create", "blobstore-backend-test"), Succeeds)
+	t.Assert(r.git("push", "flynn", "master"), Succeeds)
+
+	// set default backend to S3
+	t.Assert(flynn(t, "/", "-a", "blobstore", "env", "set", "BACKEND_S3="+s3Config, "DEFAULT_BACKEND=s3"), Succeeds)
```

Comment:
The `flynn` commands are echoed to the log, which means `s3Config` here gets printed.